assets/src/application/services/hooks/useMemoStringify.ts

```diff
@@ -0,0 +1,19 @@
+import { useMemo } from 'react';
+import { is } from 'ramda';
+
+const useMemoStringify = (data: any, deps?: any[]): any => {
+	let dataStringified: string;
+
+	if (deps && Array.isArray(deps)) {
+		dataStringified = deps.join(':');
+	}
+
+	if (is(Object, data)) {
+		dataStringified = JSON.stringify(data);
+	}
+
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	return useMemo(() => data, [dataStringified]);
+};
+
+export default useMemoStringify;
```

Comment:
I am sorry but I don't understand the idea behind `deps`. I think when the data is being stringified, that should be enough for dependency check.

Reply:
Initially, I've created this hook to handle cases like these:

![image](https://user-images.githubusercontent.com/1477453/82901011-035b0a00-9f66-11ea-8720-4aed9fcdd32c.png)

Reply:
that was needed for 2 reasons:
- to hide that `.join(':')` that was repeated multiple times
- to avoid using `// eslint-disable-next-line react-hooks/exhaustive-deps` every time a dependency is stringified.

Reply:
and after dealing with those arrays as dependency, I noticed that we have cases when object is stringified, like here:

![image](https://user-images.githubusercontent.com/1477453/82901432-92682200-9f66-11ea-92f5-99fb9dbd1ec8.png)

***
and decided to abstract this case also in `useMemoStringify`, by checking first if that is object and then applying `JSON.stringify`

Reply:
Thank you for the explanation 👍 
I have one last concern regarding this
```js
if (deps && Array.isArray(deps)) {
	dataStringified = deps.join(':');
}
```
it's also possible that `deps` may not always be an array of `strings`, it can be an array of objects. So it may be better to do this to make sure we are joining strings:
```js
import { toString } from 'ramda';

if (deps && Array.isArray(deps)) {
	dataStringified = deps.map(toString).join(':');
}
```

Reply:
thx, updated